### ISSUE TITLE
Add a confirmation option when equipping an avatar with unsaved changes

### ISSUE THREAD

**Issue by epicEaston197** [Yellow-Dog-Man/Resonite-Issues#1262]
### Is your feature request related to a problem? Please describe.

When I'm in resonate I find myself editing my avatar sometimes and I can't accidentally click equip when I meant to save my avatar and lose a lot of work so this option would be nice

### Describe the solution you'd like

When your avatar has unsaved changes have a confirmation option before you equip the avatar disclosing that you will lose all data that has been not saved

### Describe alternatives you've considered

None really except for maybe just clicking the save option instead next time and just cope with you losing hours of work?

**Comment by 5H4D0W-X** [comment]
The change tracking wouldn't be accurate because many avatars have systems that dynamically move, duplicate or destroy slots. In fact, if any change in a field is considered an unsaved change, simply having a write node or clothing toggle would count as an unsaved change. And when everything is considered an unsaved change, with time people will start skipping the warning out of habit, making it useless.

**Comment by epicEaston197** [comment]
Yeah the diagnostics on what counts as an unsaved change would have to be thought out more what I gave was mostly an examplea

**Comment by JackTheFoxOtter** [comment]
Maybe a more generic approach would be making equipping avatars undoable?

**Comment by epicEaston197** [comment]
That is interesting but don't know how that will be achieved

**Comment by Frooxius** [comment]
To add to Shifty's comments, this feels a bit like the wrong approach.

What exactly would you actually expect to be undone with this? Equipping an avatar can make a lot of things happen and a lot of them are not very nicely undoable.

The other part is, how do you equip the avatar on accident? There should be a context menu asking to confirm. Are you not getting that?

We could potentially add a way to temporarily block equipping.

**Comment by epicEaston197** [comment]
> Oh- so you're editing the avatar while you're in it, @epicEaston197...?

Yes I am I do this often 

while yes I could just spawn out a copy of my avatar it still runs into the same issue after I'm done editing my avatar externally after i spawned it out I equip it then I want to save it leading to the same issue to where I could accidentally click the equip button 

a confirmation dialogue in the inventory like 3D space is still nice to have
